src/components/layout/Button.vue

```diff
@@ -0,0 +1,19 @@
+<template>
+  <button  :class="btnClasses ? btnClasses : 'inline-flex gap-2 items-center justify-center bg-peach w-full rounded-md border border-transparent py-2 px-4 text-base font-semibold text-black shadow-sm hover:bg-peach-2 focus:outline-none focus:ring-2 focus:ring-peach focus:ring-offset-2 ease-in-out duration-300'"
+    @click="onClick">

```

Comment:
It should emit an event instead of passing a function as prop.

Reply:
Also it should be able to handle icons on both left and right of the text, using slots